rs/ethereum/cketh/minter/src/eth_logs/scraping.rs

```diff
@@ -0,0 +1,93 @@
+use crate::eth_logs::{
+    LogParser, ReceivedErc20LogParser, ReceivedEthLogParser, RECEIVED_ERC20_EVENT_TOPIC,
+    RECEIVED_ETH_EVENT_TOPIC,
+};
+use crate::eth_rpc::{FixedSizeData, Topic};
+use crate::numeric::BlockNumber;
+use crate::state::State;
+use ic_ethereum_types::Address;
+
+/// Trait for managing log scraping.
+pub trait LogScraping {
+    /// The parser type that defines how to parse logs found by this log scraping.
+    type Parser: LogParser;
+
+    fn next_scrape(state: &State) -> Option<Scrape>;
+    fn update_last_scraped_block_number(state: &mut State, block_number: BlockNumber);
+    fn display_id() -> &'static str;
+}
+
+pub struct Scrape {
+    pub contract_address: Address,
+    pub last_scraped_block_number: BlockNumber,
+    pub topics: Vec<Topic>,
+}
+
+pub enum ReceivedEthLogScraping {}
+
+impl LogScraping for ReceivedEthLogScraping {
+    type Parser = ReceivedEthLogParser;
+
+    fn next_scrape(state: &State) -> Option<Scrape> {
+        let contract_address = *state.eth_log_scraping.contract_address()?;
+        let last_scraped_block_number = state.eth_log_scraping.last_scraped_block_number();
+        let topics = vec![Topic::from(FixedSizeData(RECEIVED_ETH_EVENT_TOPIC))];
+        Some(Scrape {
+            contract_address,
+            last_scraped_block_number,
+            topics,
+        })
+    }
+
+    fn update_last_scraped_block_number(state: &mut State, block_number: BlockNumber) {
+        state
+            .eth_log_scraping
+            .set_last_scraped_block_number(block_number);
+    }
+
+    fn display_id() -> &'static str {
```

Comment:
Out of curiosity, isn't `'static` inferred automatically here?

Reply:
There it cannot be inferred because there is no method parameters with any lifetime, see https://doc.rust-lang.org/reference/lifetime-elision.html

```
// Cannot infer, because there are no parameters to infer from.
fn get_str() -> &str;                                 // ILLEGAL
```

Reply:
Closing this thread to merge. Feel free to ping me if there is still some unclarity.